cardinal/world_context.go

```diff
@@ -46,6 +50,21 @@ func NewReadOnlyWorldContext(world *World) engine.Context {
 	}
 }
 
+func GetMessage[In any, Out any](wCtx engine.Context) (*message.MessageType[In, Out], error) {
```

Comment:
As per my other comment, MessageType is an internal type and this abstraction violation and will create tight coupling.

I also saw that this is only used once for a test, so I don't see a strong need for this either.

Reply:
nvm, i understand what you are trying to do now, you still can't use `message.MessageType` though, I would make this private so you don't shoot yourself in the foot. Instead, you need to use `types.Message`.

I would also move this to cardinal.go together with `GetComponent`

Reply:
Hey so I tried this but it didn't work: 

MessageType is the current type that contains Each which is what needs to be exposed here. It's hard to put Each onto the Message interface because it contains Generic parameters. That's the issue, because it contains generic parameters it will actually pollute the Message interface if I put that function signature on there and it will then pollute everything else that uses the Message interface including Context. Etc. 

There's probably a fix around this, it will take a bit. I will keep this in mind. For now this will go up and I will change it later.

Reply:
I agree with this technical direction and I prefer it over the current method. For the record I won't be going with it due to time constraints. I think the current way actually works fine though not ideal. I talked with @jerargus (correct me if I'm wrong Jeremy) that you didn't want to block the ticket so you won't request changes. I am going to resolve the ticket and merge because of that. We can always roll back if you change your mind.